src/server/rdb_load.cc

```diff
@@ -2629,7 +2640,11 @@ void RdbLoader::LoadItemsBuffer(DbIndex db_ind, const ItemsBuf& ib) {
   }
 
   for (auto* item : ib) {
-    item_queue_.Push(item);
+    if (ShouldReuse(item->val.obj)) {
```

Comment:
note that you have another place that you push to item_queue_, in the inline flow